### PR TITLE
Engine: keep track of the nodes that correspond to a "byKey" command

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [X] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [X] Include appropriate tests
- [X] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Comment by remyhaemmerle-da** [comment]
cc @andreaslochbihler-da

**Comment by remyhaemmerle-da** [comment]
this will be plug to the participant state in a separate PR.

File: daml-lf/engine/src/test/scala/com/digitalasset/daml/lf/engine/EngineTest.scala [hunk 1]
```diff
@@ -1248,7 +1269,28 @@ class EngineTest extends WordSpec with Matchers with EitherValues with BazelRunf
...
+        "Lookup",
+        ValueRecord(None, ImmArray((Some[Name]("n"), ValueInt64(42)))))
+      val Right((tx, txMeta)) = engine
+        .submit(
+          Commands(alice, ImmArray(exerciseCmd), now, "test"),
+          participant,
+          Some(submissionSeed))
+        .consume(lookupContractMap.get, lookupPackage, lookupKey)
+
+      val fetchNodes = tx.nodes.collect {
```
Comment (gerolf-da) [inline daml-lf/engine/src/test/scala/com/digitalasset/daml/lf/engine/EngineTest.scala:1285]:
```suggestion
      val lookupByKeyNodes = tx.nodes.collect {
```

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala [hunk 2]
```diff
@@ -809,6 +817,7 @@ final case class Compiler(packages: PackageId PartialFunction Package) {
                 SEVar(3), // choice argument
                 SEVar(4), // cid
                 SEVar(5), // actors
+                SEVar(6), // byKey flag
```
Comment (gerolf-da) [inline daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala:820]: Please add the `byKey` flag to the various `// stack:` comments.

Comment (remyhaemmerle-da) [reply daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala:820]: Good catch

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala [hunk 3]
```diff
@@ -776,6 +783,7 @@ final case class Compiler(packages: PackageId PartialFunction Package) {
         Map.empty,
         0,
         withEnv { _ =>
+          env = env.incrPos // byKey flag
```
Comment (gerolf-da) [inline daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala:786]: I think `byKey` still needs to be reflected in line 777.

Comment (remyhaemmerle-da) [reply daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala:786]: Indeed. Line 775 as well.

**Pull request merged**